lib/moodle_rb/courses.rb

```diff
@@ -69,21 +69,26 @@ def create(params)
       response.parsed_response.first
     end
 
-    def show(id)
+    def show(*id)
+      ids = id.map.with_index do |item, idx|
+        [idx.to_s, item]
+      end.to_h
+
       response = self.class.post(
         '/webservice/rest/server.php',
         {
           :query => query_hash('core_course_get_courses', token),
           :body => {
             :options => {
-              :ids => {
-                '0' => id
-              }
+              id: ids
             }
           }
         }.merge(query_options)
       )
       check_for_errors(response)
+
+      return response.parsed_response if response.parsed_response.count > 1
```

Comment:
What does the Moodle API do if you call it with course IDs in the list that do not exist? Hopefully it will skip those and give us just the ones it has. If that's the case, it's possible we may only have 1, or even 0, in the response, even though the caller has passed multiple ids and is expecting an array back, so I think we should return an array (even of 0 or 1 item) if the user has passed more than one id.

Reply:
I was actually going to suggest a change to this to a ternary statement, but then got sidetracked.

I do think we should check the number of `id`s passed in as described above.